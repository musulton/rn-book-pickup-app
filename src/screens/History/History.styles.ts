import {StyleSheet, TextStyle, ViewStyle} from "react-native";

import {COLORS, SIZES} from "../../constants";

interface Styles {
    container: ViewStyle
    scrollContainer: ViewStyle
    headerTitle: TextStyle
    headerWrapper: ViewStyle
}

const styles = StyleSheet.create<Styles>({
    container: {
        display: "flex",
        flex: 1,
        backgroundColor: COLORS.background,
        paddingTop: SIZES.padding1
    },
    scrollContainer: {
        justifyContent: "center",
        alignItems: "center",
        flex: 1
    },
    headerWrapper: {
        borderBottomWidth: 1,
        borderBottomColor: COLORS.lightGray2,
        paddingVertical: SIZES.padding2,
    },
    headerTitle: {
        textAlign: "center",
        fontSize: 20,
        fontWeight: "bold",
    }
});

export default styles
