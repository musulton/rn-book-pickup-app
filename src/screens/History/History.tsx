import {FlatList, ListRenderItemInfo, SafeAreaView, Text, View} from "react-native";
import React from "react";
import {useSelector} from "react-redux";

import {Book} from "../../types";
import CardBook from "../../components/CardBook";
import Separator from "../../components/Separator";

import styles from "./History.styles";
import EmptyState from "../../components/EmptyState";

const History: React.FC = () => {
    const books = useSelector((state: any) => state?.books)

    return (
        <SafeAreaView style={styles.container}>
            <View style={styles.headerWrapper}>
                <Text style={styles.headerTitle}>List of books borrowed</Text>
            </View>
            <FlatList
                data={books}
                renderItem={(props: ListRenderItemInfo<Book>) => (
                    <CardBook {...props.item}/>
                )}
                ItemSeparatorComponent={Separator}
                contentContainerStyle={books.length < 1 && styles.scrollContainer}
                ListEmptyComponent={<EmptyState text={"You haven't picked up any books yet"} />}
            />
        </SafeAreaView>
    );
}

export default History
