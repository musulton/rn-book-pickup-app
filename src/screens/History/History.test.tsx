import {useSelector} from "react-redux";
import {cleanup, render, screen} from "@testing-library/react-native";

import History from "./History";
import {book} from "../../fixtures/testData";

jest
    .mock("react-redux")
    .mock("../../components/EmptyState", () => "EmptyState")
    .mock("../../components/CardBook", () => "CardBook")
    .mock("../../components/Separator", () => "Separator")

describe("<History />", () => {
    beforeEach(() => {
        cleanup()
    })

    it("should render History correctly", () => {
        useSelector.mockReturnValue([ book ])
        render(<History />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    it("should render History correctly when empty state", () => {
        useSelector.mockReturnValue([])

        render(<History />)

        expect(screen.toJSON()).toMatchSnapshot()
    })
})
