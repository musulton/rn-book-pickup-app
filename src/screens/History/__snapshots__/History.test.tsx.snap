// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<History /> should render History correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FEFEFE",
      "display": "flex",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      {
        "borderBottomColor": "#EFEFF0",
        "borderBottomWidth": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      List of books borrowed
    </Text>
  </View>
  <RCTScrollView
    ItemSeparatorComponent="Separator"
    ListEmptyComponent={
      <EmptyState
        text="You haven't picked up any books yet"
      />
    }
    contentContainerStyle={false}
    data={
      [
        {
          "author": "author",
          "edition": "edition",
          "title": "title",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <CardBook
          author="author"
          edition="edition"
          title="title"
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`<History /> should render History correctly when empty state 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#FEFEFE",
      "display": "flex",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      {
        "borderBottomColor": "#EFEFF0",
        "borderBottomWidth": 1,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      List of books borrowed
    </Text>
  </View>
  <RCTScrollView
    ItemSeparatorComponent="Separator"
    ListEmptyComponent={
      <EmptyState
        text="You haven't picked up any books yet"
      />
    }
    contentContainerStyle={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <EmptyState
        onLayout={[Function]}
        text="You haven't picked up any books yet"
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
